### PR TITLE
#624 Profile Type not saving when Wordpress Role is not assigned

### LINKED ISSUE #624: Profile Type not saving when Wordpress Role is not assigned
**Describe the bug**
Users who edit their profile on the frontend could not save the Profile Type synchronously on the backend. Profile Type is not saving when there is no Wordpress role is associated with it.

**To Reproduce**
1. Create a Profile Type, with Wordpress Roles set to none
2. Create a Fieldset, and Select Profile Type
3. On a different user with subscriber role (or any role), try to edit your profile and select a Profile Type
4. When you have selected a Profile Type on the frontend edit and saved it, the back end is not updated
Please watch the attached video for replication

**Expected behavior**
The update on the front end should also synch with the backend

**Screenshots**
https://drive.google.com/open?id=1CWxDeBayPIwoUxY86bfwhwKMM4pq4-mg

**Support ticket links**
https://buddyboss.zendesk.com/agent/tickets/63386

### PR DESCRIPTION
**Pull request by chetansatasiya**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #624 .

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-xprofile/screens/edit.php [hunk 1]
```diff
@@ -80,34 +80,38 @@ function xprofile_screen_edit_profile() {
 				// Get selected profile type role.
 				$selected_member_type_wp_roles = get_post_meta( $_POST[ 'field_' . $field_id ], '_bp_member_type_wp_roles', true );
 
-				if (
-					! empty( $selected_member_type_wp_roles )
-					&& 'none' !== $selected_member_type_wp_roles[0]
-				) {
-
-					if ( 'administrator' !== $selected_member_type_wp_roles[0] && current_user_can( 'administrator' ) ) {
+				if ( current_user_can( 'administrator' ) ) {
```
Comment (mehulkaklotar) [inline src/bp-xprofile/screens/edit.php]:
```suggestion
				if ( bp_current_user_can( 'administrator' ) ) {
```

File: src/bp-xprofile/screens/edit.php [hunk 2]
```diff
@@ -80,34 +80,38 @@ function xprofile_screen_edit_profile() {
...
+					if ( 'none' === $selected_member_type_wp_roles[0] ) {
+						bp_set_member_type( bp_displayed_user_id(), '' );
+						bp_set_member_type( bp_displayed_user_id(), $member_type_name );
+					} elseif ( 'administrator' !== $selected_member_type_wp_roles[0] ) {
 						$errors                  = true;
 						$bp_error_message_string = __( 'Changing this profile type would remove your Administrator role and lock you out of the WordPress admin.', 'buddyboss' );
 						$validations[]           = $bp_error_message_string;
-					} elseif ( ! in_array( $selected_member_type_wp_roles[0], array( 'editor', 'administrator' ) ) && current_user_can( 'editor' ) ) {
+					}
+				} elseif ( current_user_can( 'editor' ) ) {
```
Comment (mehulkaklotar) [inline src/bp-xprofile/screens/edit.php]:
```suggestion
				} elseif ( bp_current_user_can( 'editor' ) ) {
```

File: src/bp-xprofile/screens/edit.php (hunk 1 above)
Comment (chetansatasiya) [reply src/bp-xprofile/screens/edit.php]: @mehulkaklotar did the changes.

File: src/bp-xprofile/screens/edit.php (hunk 2 above)
Comment (chetansatasiya) [reply src/bp-xprofile/screens/edit.php]: @mehulkaklotar did the changes.

**Pull request merged**